extensions/quartz/runtime/src/main/java/io/quarkus/quartz/runtime/QuartzScheduler.java

```diff
@@ -323,6 +323,28 @@ private Properties getSchedulerConfigurationProperties(QuartzSupport quartzSuppo
             }
         }
 
+        buildTimeConfig.triggerListeners.forEach((key, value) -> {
+            props.put(StdSchedulerFactory.PROP_TRIGGER_LISTENER_PREFIX + "." + key + ".class", value.clazz);
+            value.propsValue
+                    .forEach((s, s2) -> props
+                            .put(String.format("%s.%s.%s", StdSchedulerFactory.PROP_TRIGGER_LISTENER_PREFIX, key, s), s2));
+        });
+
+        buildTimeConfig.jobListeners.forEach((key, value) -> {
+            props.put(StdSchedulerFactory.PROP_JOB_LISTENER_PREFIX + "." + key + ".class", value.clazz);
+            value.propsValue
+                    .forEach((s, s2) -> props
+                            .put(String.format("%s.%s.%s", StdSchedulerFactory.PROP_JOB_LISTENER_PREFIX, key, s), s2));
+        });
+
+        buildTimeConfig.plugins.forEach((key, value) -> {
+            props.put(StdSchedulerFactory.PROP_PLUGIN_PREFIX + "." + key + ".class", value.clazz);
+            value.propsValue
+                    .forEach(
+                            (s, s2) -> props.put(String.format("%s.%s.%s", StdSchedulerFactory.PROP_PLUGIN_PREFIX, key, s),
+                                    s2));
+        });
```

Comment:
We tend to avoid usage of lambdas in runtime modules. Can we go for classic for loops here? Probably extracting this logic into a method `addListenerProperty(String prefix, QuartzAdditionalPropsConfig config) {...}`, where the prefix is one of 
- `StdSchedulerFactory.PROP_PLUGIN_PREFIX` , 
- `StdSchedulerFactory.PROP_JOB_LISTENER_PREFIX` and 
-`StdSchedulerFactory.PROP_JOB_LISTENER_PREFIX` 

would remove some of the duplication.

Reply:
I did the refactoring. It's much better this way, without the duplication

```java
        props.putAll(getAdditionalConfigurationProperties(StdSchedulerFactory.PROP_PLUGIN_PREFIX, buildTimeConfig.plugins));
        props.putAll(getAdditionalConfigurationProperties(StdSchedulerFactory.PROP_JOB_LISTENER_PREFIX,
                buildTimeConfig.jobListeners));
        props.putAll(getAdditionalConfigurationProperties(StdSchedulerFactory.PROP_TRIGGER_LISTENER_PREFIX,
                buildTimeConfig.triggerListeners));

        return props;
    }

    private Properties getAdditionalConfigurationProperties(String prefix, Map<String, QuartzAdditionalPropsConfig> config) {
        Properties props = new Properties();
        for (String key : config.keySet()) {
            props.put(String.format("%s.%s.class", prefix, key), config.get(key).clazz);
            for (String propsName : config.get(key).propsValue.keySet()) {
                props.put(String.format("%s.%s.%s", prefix, key, propsName),
                        config.get(key).propsValue.get(propsName));
            }
        }
        return props;
    }
```